external-crates/move/crates/move-vm-runtime/src/runtime/package_resolution.rs

```diff
@@ -27,16 +25,57 @@ use std::{
     sync::Arc,
 };
 
+// Retrieves a package from the cache, attempting to load it from the data store if
+// it is not present.
+pub fn resolve_package(
+    cache: &MoveCache,
+    module_resolver: impl ModuleResolver,
+    package_to_read: VersionId,
+) -> VMResult<Option<Arc<move_cache::Package>>> {
+    let data_store = TransactionDataCache::new(module_resolver);
+    let mut packages = resolve_packages(cache, &data_store, BTreeSet::from([package_to_read]))?;
+
+    if packages.is_empty() {
+        return Ok(None);
+    }
+
+    let Some(pkg) = packages.remove(&package_to_read) else {
+        debug_assert!(false, "A different package was loaded than was requested");
+        return Err(
+            PartialVMError::new(StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR)
+                .with_message(
+                    "Package not found in loaded cache despite just loading it".to_string(),
+                )
+                .finish(Location::Undefined),
+        );
+    };
+
+    debug_assert!(
+        packages.is_empty(),
+        "More than one package was loaded when only one was requested"
+    );
+    if !packages.is_empty() {
+        error!("[VM] More than one package was loaded when only one was requested: {packages:#?}");
+        return Err(
+            PartialVMError::new(StatusCode::UNKNOWN_INVARIANT_VIOLATION_ERROR)
+                .with_message(
+                    "More than one package was loaded when only one was requested".to_string(),
+                )
+                .finish(Location::Undefined),
+        );
+    }
+
+    Ok(Some(pkg))
+}
+
 // Retrieves a set of packages from the cache, attempting to load them from the data store if
 // they are not present.
 pub fn resolve_packages(
     cache: &MoveCache,
-    natives: &NativeFunctions,
```

Comment:
I appreciate this change because the natives are also held on the cache, but it makes me wonder if we should remove them from there instead of from here.

Moreover, it does not appear the cache's natives were ever used until now. I think this makes the case even stronger for removing them from the cache, and simply letting the runtime hold a copy. They are only provided here to ensure all the required natives are defined during verification.

Reply:
Mmm very fair. I'll make a followup PR removing them from the cache, and replumbing things from there